public/components/metrics/index.tsx

```diff
@@ -77,7 +77,7 @@ export const Home = ({
   const dispatch = useDispatch();
 
   const onSelectedDataSource = async (dataSources: DataSourceOption[]) => {
-    const id = dataSources[0] ? dataSources[0].id : '';
+    const id = dataSources[0] ? dataSources[0].id : null;
```

Comment:
Only one question: What's the condition when `dataSources[0]` will be undefined and we have to set the id as `null`?

Reply:
The only condition this should be triggered is when a workspace is created and its datasource is removed putting it in a state where it has no data sources available at all. In this case metrics won't work and we don't want to display options. 
We should probably have an empty state for this such as " No connected data source with redirection to associate one".